public/js/app.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+let lastSubreddit = '';
+
+function getData(subreddit) {
+  lastSubreddit = subreddit;
+  let posts = '';
+
+  function subRedditReqListener() {
+    content.innerHTML = '';
+
+    posts = JSON.parse(this.responseText).data.children;
+
+    let contentHolder = document.createElement('div');
+    contentHolder.className = 'container';
+    content.appendChild(contentHolder);
+
+    posts.forEach((post) => {
+      let subContent = document.createElement('div');
+      subContent.className = 'subContent';
+      contentHolder.appendChild(subContent);
+
+      subContent.addEventListener('click', goToLink);
+
+      function goToLink() {
+        window.open(post.data.url, '_blank');
+      }
+
+      if (post.data.thumbnail !== 'self') {
+        let subImg = document.createElement('div');
+        subImg.className = 'subImg';
+        subContent.appendChild(subImg);
+
+        let imgHere = document.createElement('img');
+        imgHere.className = 'redditImage';
+        imgHere.src = post.data.thumbnail;
+        imgHere.onerror = function() {
+          imgHere.src = 'http://trifectaecosystems.com/wp-content/uploads/2015/04/snoo-300x208.jpg';
+        };
+        subImg.appendChild(imgHere);
+      }
+
+      let titleDiv = document.createElement('div');
+      titleDiv.className = 'title';
+      titleDiv.innerHTML = post.data.title;
+      subContent.appendChild(titleDiv);
+
+      let statsDiv = document.createElement('div');
+      statsDiv.className = 'stats';
+      subContent.appendChild(statsDiv);
+
+      let author = document.createElement('div');
+      author.className = 'author statsItem';
+      author.innerHTML = post.data.author;
+      statsDiv.appendChild(author);
+
+      let dot = document.createElement('div');
+      dot.className = 'statDot statsItem';
+      dot.innerHTML = '&#8226';
+      statsDiv.appendChild(dot);
+
+      function timeSince(date) {
+        return moment(date).fromNow();
+      }
+
+      let timelapse = document.createElement('div');
+      timelapse.className = 'timelapse statsItem';
+      timelapse.innerHTML = timeSince(post.data.created * 1000);
+      statsDiv.appendChild(timelapse);
+
+      let dot2 = document.createElement('div');
+      dot2.className = 'statDot statsItem';
+      dot2.innerHTML = '&#8226';
+      statsDiv.appendChild(dot2);
+
+      let score = document.createElement('div');
+      score.className = 'score statsItem';
+      score.innerHTML = `${post.data.score} upvotes`;
+      statsDiv.appendChild(score);
+
+      let selfText = document.createElement('div');
+      selfText.className = 'selfText';
+      selfText.innerHTML = post.data.selftext;
+      subContent.appendChild(selfText);
+    });
+  }
+
+  if (!posts) {
```

Comment:
To fix your bug with the error showing before your subreddit content shows, move this if statement to right after line 12. Then check to see if the response data has any children. If there are no children, then you should be showing the error. If there are children, then you can continue with the rest of the function. The `posts` variable isn't set until after your call to get the subreddit data comes back from the server. So the way it is right now, when line 88 fires, it's always going to be undefined.  Because lines 9 through 86 haven't actually fired yet but will when line 96 is done.

Reply:
I was stuck on this for a while.  If the subreddit doesn't exist, I get two notices: 

Access to XMLHttpRequest at 'https://www.reddit.com/r/fgdfgdf' from origin 'http://127.0.0.1:59989' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

and 

VM224:1 Cross-Origin Read Blocking (CORB) blocked cross-origin response https://www.reddit.com/r/fgdfgdf with MIME type text/html. See https://www.chromestatus.com/feature/5629709824032768 for more details.

However, there is no response text or status to reference.   If I console log 'test' in the request listener and type in a nonexistent subreddit, it doesn't seem to display 'test' in the console..